### PR TITLE
🧿 Modals for Ajna Borrow

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# [Modals for Ajna Borrow](https://app.shortcut.com/oazo-apps/story/8965/borrow-add-modals)

Only modals, no tooltips as there must have been a misunderstanding with tooltips - their content and titles doesn't match what we currently have in footer so they are probably are supposed to be in some other place, than we anticipated while estimating the story.

image
  
## Changes 👷‍♀️

- Added modals to overview section cards for Ajna Borrow,
- created helper component `DetailsSectionContentSimpleModal` for all default looking modals* - that allowed to get rid of a lot of duplication previously needed to handle all modals,
- created Ajna specific helper `AjnaDetailsSectionContentSimpleModal` for `DetailsSectionContentSimpleModal` that wraps modals in Ajna theme and applies appropriate colors.

* `ContentCardPositionLendingPrice` is an example of card then modal is more complicated than default. In that case, separated component for modal content is still needed, and `DetailsSectionContentSimpleModal` serves just as a wrapper.
  
## How to test 🧪

- See modals on any Ajna Borrow position,
- see if preexisting modals on Ajna Earn positions didn't break.

### CODE REVIEW

File: components/DetailsSectionContentSimpleModal.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { PropsWithChildren } from 'react'
+import { Card, Grid, Heading, Text } from 'theme-ui'
+
+export interface DetailsSectionContentSimpleModalProps {
+  description?: string
+  title: string
```
Comment (piekczyk) [inline components/DetailsSectionContentSimpleModal.tsx:6]: IMO it's much nicer if required props are first in the interface, not required after.

Comment (piekczyk) [reply components/DetailsSectionContentSimpleModal.tsx:6]: Take a look how hints looks when optional parameter is declared first and required one later vs opposite.

image

vs

image

File: features/ajna/common/components/AjnaDetailsSectionContentSimpleModal.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {
+  DetailsSectionContentSimpleModal,
+  DetailsSectionContentSimpleModalProps,
+} from 'components/DetailsSectionContentSimpleModal'
+import React, { PropsWithChildren } from 'react'
+import { ajnaExtensionTheme } from 'theme'
+import { ThemeProvider } from 'theme-ui'
+
+export const AjnaDetailsSectionContentSimpleModal = (
```
Comment (piekczyk) [inline features/ajna/common/components/AjnaDetailsSectionContentSimpleModal.tsx:9]: What about utilizing `FC`? Btw. when using FC you don't need to use `PropsWithChildren`.

File: components/DetailsSectionContentSimpleModal.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  value?: string
+}
+
+export function DetailsSectionContentSimpleModal({
```
Comment (piekczyk) [inline components/DetailsSectionContentSimpleModal.tsx:10]: maybe it's worth to convert this component to arrow function with `FC`

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: components/DetailsSectionContentSimpleModal.tsx (hunk 1 above)
Comment (piotrkonowrocki) [reply components/DetailsSectionContentSimpleModal.tsx:5]: Fixed.

File: components/DetailsSectionContentSimpleModal.tsx (hunk 3 above)
Comment (piotrkonowrocki) [reply components/DetailsSectionContentSimpleModal.tsx]: And fixed.

**Review by piekczyk — APPROVED** [review]

File: features/ajna/common/components/AjnaDetailsSectionContentSimpleModal.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply features/ajna/common/components/AjnaDetailsSectionContentSimpleModal.tsx]: Fixed.

**Pull request merged**